commons/src/main/java/org/sonarsource/analyzer/commons/BuiltInQualityProfileReader.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SonarQube Analyzer Commons
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.analyzer.commons;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.sonar.api.server.profile.BuiltInQualityProfilesDefinition.NewBuiltInQualityProfile;
+
+/**
+ * Use to create {@link NewBuiltInQualityProfile} based on json file, for profiles implementing {@link org.sonar.api.server.profile.BuiltInQualityProfilesDefinition}
+ *
+ * Not designed for multi-threads
+ */
+public final class BuiltInQualityProfileReader {
+
+  private NewBuiltInQualityProfile profile;
+
+  public BuiltInQualityProfileReader(NewBuiltInQualityProfile profile) {
```

Comment:
The separation between the constructor and `activateRulesFromJson` seems a bit artificial. Maybe we could move the `profile` parameter to `activateRulesFromJson`and make `activateRulesFromJson` static.

Reply:
ok